packages/@aws-cdk/aws-pipes-enrichments-alpha/lib/api-destination.ts

```diff
@@ -0,0 +1,86 @@
+import { EnrichmentParametersConfig, IEnrichment, IPipe, InputTransformation } from '@aws-cdk/aws-pipes-alpha';
+import { IApiDestination } from 'aws-cdk-lib/aws-events';
+import { IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { CfnPipe } from 'aws-cdk-lib/aws-pipes';
+
+/**
+ * Properties for a ApiDestinationEnrichment
+ */
+export interface ApiDestinationEnrichmentProps {
+  /**
+   * The input transformation for the enrichment
+   * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-pipes-input-transformation.html
+   * @default - None
+   */
+  readonly inputTransformation?: InputTransformation;
+
+  /**
+   * The headers that need to be sent as part of request invoking the EventBridge ApiDestination.
+   *
+   * @default - none
+   */
+  readonly headerParameters?: Record<string, string>;
+
+  /**
+   * The path parameter values used to populate the EventBridge API destination path wildcards ("*").
+   *
+   * @default - none
+   */
+  readonly pathParameterValues?: string[];
+
+  /**
+   * The query string keys/values that need to be sent as part of request invoking the EventBridge API destination.
+   *
+   * @default - none
+   */
+  readonly queryStringParameters?: Record<string, string>;
+}
+
+/**
+ * An API Destination enrichment for a pipe
+ */
+export class ApiDestinationEnrichment implements IEnrichment {
+  public readonly enrichmentArn: string;
+
+  private readonly inputTransformation?: InputTransformation;
+  private readonly headerParameters?: Record<string, string>;
+  private readonly pathParameterValues?: string[];
+  private readonly queryStringParameters?: Record<string, string>;
+
+  constructor(private readonly destination: IApiDestination, props?: ApiDestinationEnrichmentProps) {
+    this.enrichmentArn = destination.apiDestinationArn;
+    this.inputTransformation = props?.inputTransformation;
+    this.headerParameters = props?.headerParameters;
+    this.queryStringParameters = props?.queryStringParameters;
+    this.pathParameterValues = props?.pathParameterValues;
+  }
+
+  bind(pipe: IPipe): EnrichmentParametersConfig {
+
+    const httpParameters: CfnPipe.PipeEnrichmentHttpParametersProperty | undefined =
+      this.headerParameters ??
```

Comment:
Thanks for the PR @mazyu36 - overall looks good to me. I'm just wondering about the assignment of the `httpParameters` variable here - in practice, does it make sense to pass in HTTP parameters if one of the parameter values is undefined? (It might be nice to add a test for this, too?)

Reply:
@sumupitchayan 
Yes, I think it has meaning. For example, when calling an API that only requires path parameters. 
This is in line with similar implementations as follows:
https://github.com/aws/aws-cdk/blob/main/packages/@aws-cdk/aws-pipes-targets-alpha/lib/api-destination.ts
https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-events-targets/lib/api-destination.ts

I think it depends on the design of the API being called, so I think it would be sufficient to confirm that the parameters are set in the unit test. 
https://github.com/aws/aws-cdk/pull/31312/files#diff-0523525438cc22991d3e0555d86f9ec9c8a79cfa4a86599cdb2ea8d4075d4d84R63

I think adding integration tests unnecessary.
What do you think?

Reply:
@mazyu36 thanks for the clarification. Looks good to me then 👍